backend/src/v2/objectstore/object_store.go

```diff
@@ -286,57 +202,99 @@ func downloadFile(ctx context.Context, bucket *blob.Bucket, blobFilePath, localF
+	if err != nil {
+		return nil, err
+	}
+	config.Credentials = creds
+	config.Region = aws.String(params.Region)
+	config.DisableSSL = aws.Bool(params.DisableSSL)
+	config.S3ForcePathStyle = aws.Bool(true)
+
+	// AWS Specific:
+	// Path-style S3 endpoints, which are commonly used, may fall into either of two subdomains:
+	// 1) s3.amazonaws.com
+	// 2) s3.<AWS Region>.amazonaws.com
+	// for (1) the endpoint is not required, thus we skip it, otherwise the writer will fail to close due to region mismatch.
+	// https://aws.amazon.com/blogs/infrastructure-and-automation/best-practices-for-using-amazon-s3-endpoints-in-aws-cloudformation-templates/
+	// https://docs.aws.amazon.com/sdk-for-go/api/aws/session/
+	if strings.ToLower(params.Endpoint) != "s3.amazonaws.com" {
+		config.Endpoint = aws.String(params.Endpoint)
+	}
 
-func MinioDefaultEndpoint() string {
-	// Discover minio-service in the same namespace by env var.
-	// https://kubernetes.io/docs/concepts/services-networking/service/#environment-variables
-	minioHost := os.Getenv("MINIO_SERVICE_SERVICE_HOST")
-	minioPort := os.Getenv("MINIO_SERVICE_SERVICE_PORT")
-	if minioHost != "" && minioPort != "" {
-		// If there is a minio-service Kubernetes service in the same namespace,
-		// MINIO_SERVICE_SERVICE_HOST and MINIO_SERVICE_SERVICE_PORT env vars should
-		// exist by default, so we use it as default.
-		return minioHost + ":" + minioPort
+	sess, err := session.NewSession(config)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create object store session, %v", err)
 	}
-	// If the env vars do not exist, we guess that we are running in KFP multi user mode, so default minio service should be `minio-service.kubeflow:9000`.
-	glog.Infof("Cannot detect minio-service in the same namespace, default to %s as MinIO endpoint.", defaultMinioEndpointInMultiUserMode)
-	return defaultMinioEndpointInMultiUserMode
+	return sess, nil
 }
 
-func getMinioCredential(ctx context.Context, clientSet kubernetes.Interface, namespace string) (cred *credentials.Credentials, err error) {
+func getS3BucketCredential(
+	ctx context.Context,
+	clientSet kubernetes.Interface,
+	namespace string,
+	secretName string,
+	bucketSecretKeyKey string,
+	bucketAccessKeyKey string,
+) (cred *credentials.Credentials, err error) {
 	defer func() {
 		if err != nil {
 			// wrap error before returning
-			err = fmt.Errorf("Failed to get MinIO credential from secret name=%q namespace=%q: %w", minioArtifactSecretName, namespace, err)
+			err = fmt.Errorf("Failed to get Bucket credentials from secret name=%q namespace=%q: %w", secretName, namespace, err)
 		}
 	}()
 	secret, err := clientSet.CoreV1().Secrets(namespace).Get(
 		ctx,
-		minioArtifactSecretName,
+		secretName,
 		metav1.GetOptions{})
 	if err != nil {
 		return nil, err
 	}
-	accessKey := string(secret.Data["accesskey"])
-	secretKey := string(secret.Data["secretkey"])
+	accessKey := string(secret.Data[bucketAccessKeyKey])
```

Comment:
nil: Add a comment to describe the `bucketAccessKeyKey` value is the `key` value for `bucketAccessKey`? Otherwise it could be confusing for other developers.

Reply:
> Just some minor comments on the naming convention. Otherwise it looks good to me. From the e2e test, I think it didn't break any of the existing frontend features right? 

(addressing this question here) 

yes the goal is to not break any existing frontend features, if a user decides to leverage this feature however (i.e. create the kfp-launcher config with path prefixes) those will not render, and an error will be displayed. Otherwise everything should remain the same as before, if reviewers find otherwise let me know and I'll address it.

Reply:
@Tomcli I've added comments wherever the "keykey" naming is declared for further clarification on what these variables are used for